File: Hacker Rank/Diagonal Difference/C/Diagonal_difference.c
// coding section:

#include<stdio.h>
#include<stdlib.h>

int main()
{ 
  int arr[100][100], n, i, j, left_sum = 0, right_sum = 0, result;
  scanf("%d", &n);
  for (i = 0; i < n; i++)
  {
    for (j = 0; j < n; j++)
    {
      scanf("%d", &arr[i][j]);
      if (i == j)
      {
        left_sum += arr[i][j];
      }
      if (i + j == n - 1)
      {
        right_sum += arr[i][j];
      }
    }
  }

  result = abs(left_sum - right_sum);
  printf("%d", result);

  return 0;
}

File: Hacker Rank/Diagonal Difference/Python/Diagonal Difference.py
import math
import os
import random
import re
import sys


def diagonalDifference(arr):
    sum1 = sum([arr[x][x] for x in range(n)]) 
    sum2 = sum([arr[x][n-1-x] for x in range(n)])
    return abs(sum1-sum2)

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())

    arr = []

    for _ in range(n):
        arr.append(list(map(int, input().rstrip().split())))

    result = diagonalDifference(arr)

    fptr.write(str(result) + '\n')

    fptr.close()
